tests/tests_pytorch/accelerators/test_tpu.py

```diff
@@ -323,3 +323,44 @@ def test_trainer_config_device_ids(devices, expected_device_ids):
     trainer = Trainer(accelerator="tpu", devices=devices)
     assert trainer.device_ids == expected_device_ids
     assert trainer.num_devices == len(expected_device_ids)
+
+
+@RunIf(tpu=True)
+@mock.patch.dict(os.environ, os.environ.copy(), clear=True)
+def test_all_gather(tmpdir):
```

Comment:
There's no value in adding this test, as this is all covered by the one you added in the other file

Reply:
This test is for the code modifs in the `TPUSpawnStrategy` of `pytorch_lightning` whereas the other test is for the code modifs in the `XLAStrategy` of `lightning_lite`. It concerns different classes, therefore, there should be a test for both execution paths even though the test code is the same. Do you want to imply that internally only one of those strategies are used?

It looks like the `TPUSpawnStrategy` of `pytorch_lightning` and the `XLAStrategy` of `lightning_lite` are very much the same. As I don't know much about the internal usages, I'm not well placed to say whether they can be entirely merged. Surely, efforts could be made to avoid redundancy in these two classes. However, this may best be done in a separate ticket to avoid polluting this PR.

Reply:
I generally agree with @carmocca. 
The test in the other file for lite is great, but even in that one, one could argue that it is going too far in that it is testing the correctness of `xf.all_gather`. The real test for us we care about is whether the "wrapper" method calls the underlying function correctly (with the right arguments).

Regarding the two classes:
The current state is that the two classes are independent, yes. The idea however is that in the future, the tpu strategy in pytorch_lightning will use most of the code in the other strategy in lightning_lite. 

I am ok with adding these two "slow" tests. But remember, launching processes is extremely expensive. 
If more operations get the "grad" support, we should extend this test instead of launching new ones.